File: iSeeCars.Business/Services/InputService.cs
using FluentValidation;
using FluentValidation.Results;
using iSeeCars.Business.Helpers;
using iSeeCars.Business.Managers;
using iSeeCars.Business.Validators;
using iSeeCars.Core.Entities;

namespace iSeeCars.API.Services
{
    internal class InputService
    {
        private readonly ModelManager modelManager;
        private readonly CarManager carManager;
        private readonly IValidator<Car> carValidator;

        public InputService(ModelManager modelManager,CarManager carManager,IValidator<Car> carValidator)
        {
            this.modelManager = modelManager;
            this.carManager = carManager;
            this.carValidator = carValidator;
        }

        public Car? AddCarFromUser()
        {
            while (true)
            {
                Car car = new Car();

            model:
                Console.WriteLine("Please choose a model from the list below:");

                var models = modelManager.GetAllModels();

                if (models.Count == 0)
                {
                    Console.WriteLine("Models not found. Do you wanna add a new model? (Y/N)");

                    var click = Console.ReadKey(true);

                    if (click.Key == ConsoleKey.Y)
                    {
                        var newModel = modelManager.AddModel();
                        car.ModelId = newModel.ModelId;
                    }
                    else if (click.Key == ConsoleKey.N)
                    {
                        Console.WriteLine("Operation cancelled");
                        return null;
                    }
                    else
                    {
                        Console.WriteLine("Please press Y or N");
                        goto model;
                    }
                }
                else
                {
                    modelManager.ShowAllModels();

                    int modelId = InputHelper.ReadInt(
                        "Enter a model ID",
                        mdlId => models.Any(m => m.ModelId == mdlId),
                        "Invalid model Id");

                    car.ModelId = modelId;
                }

                car.Price = InputHelper.ReadDouble(
                    "Enter price:",
                    v => v > 0,
                    "Invalid price");

                car.Color = InputHelper.ReadString(
                    "Enter color:",
                    v => !string.IsNullOrWhiteSpace(v),
                    "Invalid color");

                InputHelper.ShowEnum<FuelType>();

                int fuel = InputHelper.ReadInt(
                    "Select fuel:",
                    v => Enum.IsDefined(typeof(FuelType), v),
                    "Invalid fuel");

                car.FuelType = (FuelType)fuel;

                car.Engine = InputHelper.ReadDouble(
                    "Enter engine:",
                    v => v > 0,
                    "Invalid engine");

                int year = InputHelper.ReadInt(
                    "Enter year:",
                    v => v >= 1900 && v <= DateTime.Now.Year,
                    "Invalid year");

                car.Year = year;

                ValidationResult result = carValidator.Validate(car);

                if (result.IsValid)
                    return car;

                Console.ForegroundColor = ConsoleColor.Red;

                Console.WriteLine("Validation errors:");

                foreach (var error in result.Errors)
                    Console.WriteLine(error.ErrorMessage);

                Console.ResetColor();

                Console.WriteLine("Please try again...\n");
            }
        }
        public int ReadCarFromUser(string message)
        {

            return InputHelper.ReadInt(message, id => carManager.GetAllCars().Any(x => x.CarId == id),
                "Car not found");
            
        }
    }
}

File: iSeeCars.java/src/main/java/iseecars/util/InputService.java
package iseecars.service;
import static java.lang.System.out;

import iseecars.helper.InputHelper;
import iseecars.manager.CarManager;
import iseecars.manager.ModelManager;
import iseecars.model.Car;
import iseecars.model.FuelType;
import iseecars.model.Model;

import java.util.List;

public class InputService {
    private final ModelManager modelManager;
    private final CarManager carManager;

    public InputService(ModelManager modelManager, CarManager carManager){

        this.modelManager = modelManager;
        this.carManager = carManager;
    }
    public Car addCarFromUser(){
        Car car = new Car();
        List<Model> models = modelManager.getAllModels();

        if(models.isEmpty()){
            out.println("Models not found. Add new model first.");
            return null;
        }

        modelManager.showAllModels();

        int modelId = InputHelper.readInt(
            "Enter model Id:",
            id -> models.stream().anyMatch(m -> m.getModelId() == id),
            "Invalid model id"
        );
        car.setModelId(modelId);
        car.setPrice(InputHelper.readDouble(
                "Enter price:",
                v -> v > 0,
                "Invalid price"
        ));
        car.setColor(InputHelper.readString(
                "Enter color:",
                v -> !v.isBlank(),
                "Invalid color"
        ));
        for (FuelType fuelType : FuelType.values()) {
            System.out.println(fuelType.getValue() + ". " + fuelType);
        }

        int fuel = InputHelper.readInt(
                "Select fuel type from the list below:",
                v -> v >= 1 && v <= FuelType.values().length,
                "Invalid fuel type"
        );

        car.setFuelType(FuelType.values()[fuel - 1]);

        car.setEngine(InputHelper.readDouble(
                "Enter engine:",
                v -> v > 0,
                "Invalid engine"
        ));

        car.setYear(InputHelper.readInt(
                "Enter year:",
                v -> v >= 1900 && v <= 2026,
                "Invalid year"
        ));

        return car;

    }

    public int readCarFromUser(String message){
        return InputHelper.readInt(
                message,
                id -> carManager.getAllCars().stream().anyMatch(c -> c.getCarId() == id),
                "Car not found"
        );
    }
}
